File: esports-backend/src/main/java/com/esports/entity/MilestonePayment.java
package com.esports.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "milestone_payments")
@Data
@NoArgsConstructor
public class MilestonePayment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @Column(name = "milestone_index")
    private Integer milestoneIndex;

    @Column(name = "milestone_title")
    private String milestoneTitle;

    @Column(name = "amount")
    private Double amount;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private PaymentStatus status = PaymentStatus.PENDING;

    @Column(name = "released_at")
    private LocalDateTime releasedAt;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    public enum PaymentStatus {
        PENDING, RELEASED
    }
}

File: esports-backend-dotnet/Upkart1/Entities/MilestonePayment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Upkart1.Entities.Enums;

namespace Upkart1.Entities
{
    /// <summary>
    /// Matches esports-backend entity: MilestonePayment (table: milestone_payments).
    /// Java uses inner enum PaymentStatus: PENDING, RELEASED.
    /// </summary>
    [Table("milestone_payments")]
    public class MilestonePayment
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("project_id")]
        public long ProjectId { get; set; }
        public Project Project { get; set; } = null!;

        [Column("milestone_index")]
        public int MilestoneIndex { get; set; }

        [Column("milestone_title")]
        public string? MilestoneTitle { get; set; }

        [Column("amount")]
        public double Amount { get; set; }

        [Column("status")]
        public PaymentStatus Status { get; set; } = PaymentStatus.PENDING;

        [Column("released_at")]
        public DateTime? ReleasedAt { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
